Clear problems setting

mseiler025 opened the pull request:
added setting that if set to true will clear the problems (linter warnings/errors) for a document when you close it

mseiler025 committed:
clear problems on document close

mseiler025 committed:
adjust indentation

mseiler025 committed:
add setting for clearing problems (warnings/errors) on document close

mseiler025 committed:
change let to const

cfjedimaster commented:
Your changes should be live - I'd love a sanity check.

mseiler025 commented:
I tested it out with a clean install and getting the following error:
"The CSSLint Language Client server crashed 5 times in the last 3 minutes. The server will not be restarted."

In the "OUTPUT" tab for CSSLint Lanugage Client (in VS Code), its showing this error:
"ReferenceError: require2 is not defined"

There is a line that is now in \.vscode\extensions\raymondcamden.csslint-0.0.5\server\server.js that reads:
let foo = require2('./csslint');

I commented out that line and am no longer getting that error, however the code added to clear problems on document close is not showing in the server.js file and that functionality is not working as expected (since the code is not there).

cfjedimaster commented:
Ugh I apologize - I think it was a bad merge. I'll try to clear stuff up asap.

cfjedimaster commented:
Weird - do you see that in code? I don't. Either here or locally where I packaged.

cfjedimaster commented:
Ok, I've done some major updates (based on https://github.com/Microsoft/vscode-extension-samples/tree/master/lsp-sample). Would you be willing to test this if I emailed you a copy? Well shoot, I can just update here and not publish. Hold on.

cfjedimaster commented:
Please try with the latest. (from here, not the marketplace)

mseiler025 commented:
To be clear, the new marketplace version was what I tested before and got the errors.

**Edit: Ignore everything below.  Sorry I didn't realize you had to open the "client" folder to run the extension from the debugger.  Everything is working fine (including clearProblemsOnDocumentClose) from the GitHub repo version.**

Now I set up the extension from this repo using the first 3 steps from "How to run locally" at https://github.com/Microsoft/vscode-extension-samples/tree/master/lsp-sample

I get the following error when the new window (extension test/debug window) appears:

"[C:\Users\winusername\Desktop\vscode-csslint-master]: property `engines` is mandatory and must be of type `object`"

and the extension does not work and the extension settings are not available in settings.

However, I tried just moving the "client" folder (after npm install and npm run compile) to my C:\Users\winusername\\.vscode\extensions directory and then started VS Code and the extension is working as expected and the new setting ("cssLanguageClient.clearProblemsOnDocumentClose") is working as expected.  There is still an error that is showing repeatedly in the "OUTPUT" tab for CSSLint Lanugage Client (in VS Code):

[Error - 8:07:24 PM] Request textDocument/completion failed.
  Message: Unhandled method textDocument/completion
  Code: -32601

however that does not seem to affect functionality.

I don't know why running the extension through the development method (Debug -> Launch Extension) doesn't work now.  That method worked for me earlier while I was working on adding the clearProblemsOnDocumentClose setting and testing it.

cfjedimaster commented:
Sigh - this is .... painful. When I test with Debug (Launch Client), I get a 'terminal process' error with code 1, but no more details. :\

mseiler025 commented:
I am not sure what would cause that error but I don't think its related to this extension.  I actually realized I should have put the clear problems code within "connection.onDidCloseTextDocument" instead of "documents.onDidClose" so I changed that.  I also found out what was causing the "Unhandled method textDocument/completion" error so I commented that out.  I submitted a new pull request with those changes.

mseiler025 commented:
I tried running the extension from Debug launch from a repo and its not working but I'm not getting the error you are getting.  When I run it from downloaded zip folder/files it works.

I tried moving the code from "documents.onDidClose" to "connection.onDidCloseTextDocument" and it seemed to not change the functionality.  I'm not sure which is better to use or if it doesn't matter.  I've seen other extensions use "documents.onDidClose" so maybe that's best?

I found a fix for the "Unhandled method textDocument/completion" error; commenting out/removing the following from server.ts:

completionProvider: {
  resolveProvider: true
}

cfjedimaster commented:
I noticed you removed your PR. Should I just make the change you  listed abov?

mseiler025 commented:
Yeah, I didn't test extensively, but I think it should be removed since the extension doesn't support code completion (which I think is what that is for and why the error is occurring).  I removed the PR because I didn't see a difference in the functionality, the "Unhandled method textDocument/completion" error is unrelated and a simple fix (assuming it doesn't break anything), no PR necessary.

cfjedimaster commented:
That change is pushed. I still can't freaking debug. :\

mseiler025 commented:
Ugh, annoying.  Well I just tested with a clean downloaded copy and everything seems to be working as expected.

cfjedimaster commented:
Ok, so in theory - I publish (oh, need to increment the version) and in
theory, it's good. Here goes nothing.

On Fri, Apr 6, 2018 at 1:17 PM, Matthew Seiler <notifications@github.com>
wrote:

> Ugh, annoying. Well I just tested with a clean downloaded copy and
> everything seems to be working as expected.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/cfjedimaster/vscode-csslint/pull/4#issuecomment-379334872>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAYBvHfS1eSl_Y-HVmhB-5lcyGYgdKniks5tl7EogaJpZM4StRzT>
> .
>



-- 
===========================================================================
Raymond Camden, Senior Developer Advocate at Auth0

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

cfjedimaster commented:
0.0.9 is out - but it looks like it isn't linting. :( Can you try?

mseiler025 commented:
It's not working for me either but I think I know whats wrong.  Only the "client" folder's contents need to be packaged (after compiling).  I deleted everything not in the client folder and moved all the files from the "client" folder to the main extension folder (C:\Users\winusername\\.vscode\extensions\raymondcamden.csslint-0.0.9), closed and reopened VS code and its working.

cfjedimaster commented:
omg - did they even document this?

mseiler025 commented:
I had similar issue earlier when trying to debug and follow the instructions from here: https://github.com/Microsoft/vscode-extension-samples/tree/master/lsp-sample

and it was not clear that you needed to open the "client" folder (after compiling) and debug->launch extension from there, I guess because everything gets moved to that folder after compiling it.

cfjedimaster commented:
Here is a SO question confirming it: https://stackoverflow.com/questions/45442869/how-to-publish-a-lsp-language-server-on-vscode-as-we-do-extension

To your last comment - are you saying File/Open on client, not root?

mseiler025 commented:
For this setup when you are trying to do debug->launch extension you need to File/Open Folder on "client" not root.

cfjedimaster commented:
... I am cursing so much now. ;)

So the latest works. I'm noticing [ex] in front of every message so I want to fix that. By the way, this is the bug I fixed about debugging (https://github.com/Microsoft/vscode/issues/47328). We know the answer now, but I still want to know why VSC didn't help me figure out the issue.

cfjedimaster commented:
And now I'm debugging. Sigh. Thank you, thank you, thank you.

cfjedimaster commented:
So the ex comes from this:
		diagnostics.push({
			severity:severity,
			range: {
				start: {line:issue.line-1, character:issue.col-1},
				end: {line:issue.line-1, character:issue.col-1}
			},
			message: issue.message,
			source:'ex'
		});

I think I need to change that to CSSLint.

cfjedimaster commented:
and yep - that was it. Pushing 0.0.11 now and I'm never touching this code again. ;)

mseiler025 commented:
Ah yes.  I'm glad you got everything working.  I just tested the new release 0.0.10 and its working now :)

mseiler025 commented:
Tested 0.0.11 as well, everything is working as expected.

cfjedimaster commented:
Thank you again! :)

mseiler025 commented:
Sure :)  Learned some interesting stuff about working with/debugging extensions.
